packit_service/worker/testing_farm.py

```diff
@@ -68,16 +82,12 @@ def _payload(self, build_id: int, chroot: str) -> dict:
         """
         distro, arch = self.chroot2distro_arch(chroot)
         compose = self.distro2compose(distro)
-        pr = self.project.get_pr(self.metadata.pr_id)
-        # url of the source/fork from which the PR has been created
-        from_url = pr.source_project.get_web_url()
+        fmf = {"url": self.fmf_url, "ref": self.fmf_ref}
```

Comment:
Per [TF docs](https://testing-farm.gitlab.io/api/#operation/requestsPost), the "ref" is optional so if `fmf_ref` is not set (and `fmf_url` is set), we should not send it at all, rather than sending `{"ref": None}`.